lib/saml_idp/metadata_builder.rb

```diff
@@ -29,8 +29,10 @@ def fresh
               descriptor.SingleLogoutService Binding: "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect",
                 Location: single_logout_service_redirect_location
               build_name_id_formats descriptor
-              descriptor.SingleSignOnService Binding: "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect",
+              descriptor.SingleSignOnService Binding: "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST",
                 Location: single_service_post_location
```

Comment:
Since the setting var was misnamed to `single_service_post_location` thoughts on making it do `single_service_redirect_location || single_service_post_location` then we could potentially remove that on a v1 or something? Genuinely asking, I could go multiple directions with this, but this, as is, is sort of a breaking change.

Reply:
Also apologies, I realize this is _months_ ago at this point

Reply:
I think both HTTP methods can be in the IdP metadata at the same time. And SP should choose what they prefer.  And in rails POST and Redirect will convert internally as POST (app/views/saml_idp/idp/saml_post.html.erb). So, I think it won' break the main feature.

I agree that if value not provided we should not render in metadata xml. But there is another issue I could see what if both values are not exist? That's mean IdP configuration issue, but if we validate configuration in here it will be too late. 
So, my opinion is we need another feature which validation feature configuration at least provided minimum values specified in the SAML 2.0 specification.

Reply:
Or maybe we can make optional only `single_service_redirect_location` as follows.

`unless (single_service_redirect_location.nil? || single_service_redirect_location.empty?)`

Because, SAML 2.0 specification mentioned it should be `one or more`
[SAML2.0 spec](https://www.oasis-open.org/committees/download.php/51890/SAML%20MD%20simplified%20overview.pdf) Page 5.

Reply:
I agree that they can both be there, but doesn't this change the meaning of `single_service_post_location`? Basically I'm wondering if someone upgrades the gem to get this code, doesn't change their configuration - will it still work as expected for them?

Reply:
Sounds good. Let's roll with it then.

Reply:
Let's add the bit to check if the values are present before writing it to document then we can merge? Or did you disagree with adding the presence check?

Reply:
@jphenow let me add check in here. But still I feel that we need a different approach like minimum configuration setting validation and allow it in here more dynamic way to setup multiple values. I will work on my approach later and come with a different pull request.

Reply:
@jphenow I have added a check render element only when the config value presents. Could you please look at it?